index.js

```diff
@@ -1,200 +1,26 @@
-    });
-
-    const filesToBePublished = result.files.filter(
-      (file) => file.type === "package file"
-    );
-    if (!filesToBePublished.length) {
-      throw new Error(
-        `No files detected for upload. Did you run bundling first (if needed)? Is the "files" field of eik.json correct?`
-      );
-    }
-    logger.log("Found %d file(s) to publish", filesToBePublished.length);
-  } catch (err) {
-    logger.log("Verification checks failed to complete");
-    throw err;
-  }
-};
-
-module.exports.generateNotes = async function generateNotes() {
-  if (!publishNeeded) return;
-  return `New Eik package version ${versionToPublish} published.`;
-};
-
-module.exports.prepare = async function prepare(options, context) {
-  if (!publishNeeded) return;
-  const { cwd, logger } = context;
-  if (eikjson.version !== versionToPublish) {
-    logger.log(`Version ${versionToPublish} written to eik.json`);
-    await json.writeEik({ version: versionToPublish }, { cwd });
-  }
-};
-
-module.exports.publish = async function publish(options, context) {
-  if (!publishNeeded) return;
-  const { cwd, logger } = context;
-  try {
-    const { server, name, files, type, map, out } = eikjson;
-    const result = await eik.publish({
-      name,
-      type,
-      server,
-      files,
-      cwd,
-      token: eikToken,
-      dryRun: false,
-      version: versionToPublish,
-      map,
-      out,
-    });
-
-    const filesPublished = result.files.filter((file) => file.type === "pkg");
-    logger.log(
-      `Published ${filesPublished.length} file(s) to Eik server ${eikjson.server}`
-    );
-  } catch (err) {
-    logger.log(
-      "Publish to Eik server failed. This may leave your build in a broken state. If retrying the build fails, try pushing a new commit."
-    );
-    throw err;
-  }
-};
-
-module.exports.success = async function success(options, context) {
-  if (!publishNeeded) return;
-  const { logger } = context;
-  const { server, name, type } = eikjson;
-  const url = new URL(
-    join(common.helpers.typeSlug(type), name, versionToPublish),
-    server
-  );
-  url.searchParams.set("ts", Date.now());
-  const result = await fetch(url);
-  if (result.ok) {
-    logger.log(
-      `Successfully published package ${name} (v${versionToPublish}) to ${url.href}.`
-    );
-  } else {
-    logger.log(
-      `Unable to locate package ${name} (v${versionToPublish}) at ${url.href}. It appears something went wrong during publish.`
-    );
-  }
+const verifyConditions = require("./lib/verify-conditions");
+const analyzeCommits = require("./lib/analyze-commits");
+const verifyRelease = require("./lib/verify-release");
+const generateNotes = require("./lib/generate-notes");
+const prepare = require("./lib/prepare");
+const publish = require("./lib/publish");
+const success = require("./lib/success");
+
+class State {
+  eikToken = "";
+  eikJSON = {};
+  versionToPublish = null;
+  publishNeeded = null;
+}
+
+const state = new State();
```

Comment:
Any benefits in using a class here over assigning an object with the same properties?

Reply:
Object naming in logs is about the only reason I can think of. Initially I was going to harden the object a bit but I think now I won't bother since it ended up being quite small.